addons/project/tests/test_project_flow.py

```diff
@@ -305,6 +306,20 @@ def test_task_with_no_project(self):
             except Exception as e:
                 raise AssertionError("Error raised unexpectedly while computing a field of the project ! Exception : " + e.args[0])
 
+        # tasks with no project set should only be visible to the users assigned to them
+        task_without_project.user_ids = [Command.link(self.user_projectuser.id)]
+        task_without_project.with_user(self.user_projectuser).read(['name'])
+        with self.assertRaises(AccessError):
+            task_without_project.with_user(self.user_projectmanager).read(['name'])
+
+        # Tests that tasks assigned to the current user should be in the 'inbox' stage by default
+        task = self.env['project.task'].create({
+            'name': 'Test Task!',
+            'user_ids': [Command.link(self.env.user.id)],
+        })
+        stages = task._get_default_personal_stage_create_vals(self.env.user.id)
+        self.assertEqual('Inbox', stages[0].get('name'), "tasks assigned to the current user should be in the 'inbox' stage by default")
+
```

Comment:
> I would say that the goal should be to ensure that the task gets the right default instead of `Inbox` right? And we manage that default in `_get_default_personal_stage_create_vals `. So I would adapt to:

I agree with you but there is a small change.                                                                  
     

`self.assertEqual(task.personal_stage_id.stage_id.name, stages[0].get('name'), "tasks assigned to the current user should be in the right default stage")`

Thanks

Reply:
Sure, thanks 🙂